pyccel/codegen/printing/ccode.py

```diff
@@ -1145,6 +1145,8 @@ def _new_slice_with_processed_arguments(_slice, array_size, allow_negative_index
         return Slice(start, stop, step)
 
     def _print_PyccelArraySize(self, expr):
+        if expr.arg.is_optional:
+            return '{}->shape[{}]'.format(expr.arg, expr.index)
```

Comment:
Have a look at the more general `stored_in_c_pointer`. If you can find a solution that tests this instead of `.is_optional` this would be preferred as it will help fix other things (e.g. returning multiple arrays from a function)

Reply:
Is this problem specific to `shape` or any other ndarray property?

Reply:
It happens with other properties too if they are accessed like this. I don't remember if there are any others outside of the numpy where PR

Reply:
#858 is now merged so there is now `size` as well as `shape` which has this problem